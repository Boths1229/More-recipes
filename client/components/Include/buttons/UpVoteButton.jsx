import React from 'react';
import PropTypes from 'prop-types';

/**
 * @description functional component to render upvote button
 *
 * @param {object} props
 *
 * @returns {JSX} return JSX
 */
const UpVoteButton = (props) => {
  const { upvoteRecipe, recipeId, upvotes } = props;
  return (
    <div>
      <button className="btn btn-orange mr-2" type="button">
        <i
          className="fa fa-thumbs-up fa-2x btn-white"
          aria-hidden="true"
          onClick={() => upvoteRecipe(recipeId)}
        />
      </button>
      <span className="mr-2">
        {upvotes}
      </span>
    </div>
  );
};

UpVoteButton.propTypes = {
  upvoteRecipe: PropTypes.func.isRequired,
  recipeId: PropTypes.number.isRequired,
  upvotes: PropTypes.number.isRequired
};

export default UpVoteButton;
