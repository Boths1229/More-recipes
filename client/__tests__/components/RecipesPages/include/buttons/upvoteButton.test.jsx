import React from 'react';
import { shallow } from 'enzyme';
import UpVoteButton from '../../../../../components/Include/buttons/UpVoteButton';

describe('Upvote button', () => {
  it('matches snapshot', () => {
    const output = shallow(<UpVoteButton />);
    expect(output).toMatchSnapshot();
  });
  it('calls onClick', () => {
    const props = {
      upvoteRecipe: jest.fn()
    };
    const output = shallow(<UpVoteButton {...props} />);
    const button = output.find('i[aria-hidden="true"]');
    button.simulate('click');
    expect(props.upvoteRecipe).toHaveBeenCalled();
  });
});
