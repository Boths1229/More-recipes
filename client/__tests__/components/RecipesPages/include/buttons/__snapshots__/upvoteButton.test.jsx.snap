// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Upvote button matches snapshot 1`] = `
<div>
  <button
    className="btn btn-orange mr-2 up favorite-btn"
    type="button"
  >
    <i
      aria-hidden="true"
      className="fa fa-thumbs-up fa-2x btn-white"
      onClick={[Function]}
    />
  </button>
  <span
    className="mr-2"
  />
</div>
`;
